File: View/CustomDemo/app/src/main/java/com/saltwater/viewdemo/MainActivity.java
package com.saltwater.viewdemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.saltwater.viewdemo.customView.CustomViewActivity;
import com.saltwater.viewdemo.scroll.ScrollActivity;
import com.saltwater.viewdemo.viewEvnt.EventActivity;
import com.saltwater.viewdemo.ViewBase.ViewLocationInfoActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_base).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this,ViewLocationInfoActivity.class));
            }
        });

        findViewById(R.id.btn_event).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this,EventActivity.class));
            }
        });
        findViewById(R.id.btn_scroll).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this,ScrollActivity.class));
            }
        });
        findViewById(R.id.btn_customView).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, CustomViewActivity.class));
            }
        });
    }

}

File: SharedPreferenceDemo/app/src/main/java/com/saltwater/sharedpreferencedemo/MainActivity.kt
package com.saltwater.sharedpreferencedemo

import android.content.Context
import android.content.SharedPreferences
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        saved()
        loaded()
    }
    fun saved(){
        val sharedPreferences = getSharedPreferences("user", Context.MODE_PRIVATE).edit()
        sharedPreferences.putString("name","test")
        sharedPreferences.commit()
    }

    fun loaded(){
        val sharedPreferences = getSharedPreferences("user", Context.MODE_PRIVATE)
        Log.d("sharedPreference",sharedPreferences.getString("name",""))
    }
}
